lib/tesla/adapter/gun.ex

```diff
@@ -232,17 +232,22 @@ if Code.ensure_loaded?(:gun) do
       end
     end
 
-    defp do_open_conn(uri, %{proxy: {proxy_host, proxy_port}}, gun_opts, tls_opts) do
+    defp do_open_conn(uri, %{proxy: {proxy_host, proxy_port}} = opts, gun_opts, tls_opts) do
       connect_opts =
         uri
         |> tunnel_opts()
         |> tunnel_tls_opts(uri.scheme, tls_opts)
+        |> add_proxy_auth_credentials(opts)
```

Comment:
Would be great to have auth for `socks` proxy too. [Docs](https://ninenines.eu/docs/en/gun/2.0/manual/gun/) -> socks_opts